core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -1261,6 +1279,51 @@ class KafkaController(val config: KafkaConfig,
     controlledShutdownCallback(controlledShutdownResult)
   }
 
+  private def processSkipControlledShutdownSafetyCheck(id: Int, brokerEpoch: Long, skipControlledShutdownSafetyCheckCallback: Try[Unit] => Unit): Unit = {
+    val controlledShutdownResult = Try { doSkipControlledShutdownSafetyCheck(id, brokerEpoch) }
+    skipControlledShutdownSafetyCheckCallback(controlledShutdownResult)
+  }
+
+  private def doSkipControlledShutdownSafetyCheck(id: Int, brokerEpoch: Long): Unit = {
+    if (!isActive) {
+      throw new ControllerMovedException("Controller moved to another broker. Aborting skip shutdown safety check operation.")
+    }
+
+    val cachedBrokerEpoch = controllerContext.liveBrokerIdAndEpochs(id)
+    if (brokerEpoch < cachedBrokerEpoch) {
+      val stateBrokerEpochErrorMessage = "Received skip shutdown safety check request for an old broker epoch " +
+        s"$brokerEpoch for broker $id. Current broker epoch is $cachedBrokerEpoch."
+      info(stateBrokerEpochErrorMessage)
+      throw new StaleBrokerEpochException(stateBrokerEpochErrorMessage)
+    }
+
+    if (!controllerContext.liveOrShuttingDownBrokerIds.contains(id))
+      throw new BrokerNotAvailableException(s"Broker id $id does not exist.")
+
+    controllerContext.skipShutdownSafetyCheck += (id -> brokerEpoch)
+  }
+
+  private def safeToShutdown(id: Int, brokerEpoch: Long): Boolean = {
```

Comment:
I feel we could move the skip check inside this method, but it's up to you.

Reply:
I like having it outside of safeToShutdown since 1.) it makes safeToShutdown a little simpler, and 2.) I want to always run the safety check even if it is skipped so that we have a clear record in the log for cases where it was not safe to shut down but we skipped it anyway.